libs/commons/common2.ml

```diff
@@ -3222,6 +3222,18 @@ let uncat xs file =
              pr s;
              pr "\n"))
 
+(*****************************************************************************)
+(* Machine *)
+(*****************************************************************************)
+
+let arch () =
```

Comment:
This is not portable, will not work under windows, so please move it out of Common2.ml.

Maybe you can define libs/commons/Platform.ml
but this code will need to be defined differently for each OS.

See also Sys.os_type